src/main/java/Parser.java

```diff
@@ -0,0 +1,88 @@
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+
+/**
+ * Represents a parser that checks the validity of users' inputs.
+ */
+public class Parser {
+    /**
+     * Checks validity of users' input.
+     * @param type refers to the commands of user inputs e.g., "done", "delete", "deadline"
+     * @param userInput refers to the actual input sentence from user
+     * @throws EmptyFieldException occurs when there are missing data from user input
+     * @throws InvalidInputException occurs when user input is invalid
+     * @throws ParseException occurs when user input date is invalid
+     */
+    public static void handleInput(String type, String userInput) throws
+            EmptyFieldException, InvalidInputException, ParseException {
+        if (type.equals("find")) {
```

Comment:
I like how you organise the checkings of the user input in a single file. :D
One suggestion would be to use a switch case here to make the code look neater.